### ISSUE TITLE
Need Open Sauced GitHub Login

### ISSUE THREAD

**Issue by bdougie** [open-sauced/hot#20]
### Type of feature

Feature

### Current behavior

The user profile is currently a placeholder. Need a GitHub login to confirm the following

- stars
- Open sauced goals account
- public comments

![image](https://user-images.githubusercontent.com/5713670/146656265-c4e31535-0f85-4cb8-a1f1-512108dfbff5.jpeg)

### Desired solution

We can mirror the logging from github.com/bdougie/live or github.com/open-sauced/open-sauced

1. Add login functionally to the Astro app
2. Replaces the Generic Avatar with the logged in user Avatar

### Additional context

The open sauced login is powered by a GitHub App Id 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bdougie** [comment]
Doing some light research and think Netlify with is the way to go here.

**Comment by 0-vortex** [comment]
<img width="747" alt="Screenshot 2021-12-19 at 03 30 23" src="https://user-images.githubusercontent.com/237133/146660401-01ae2974-a574-46dc-b544-4d48b16f97ba.png">

Have used it in some demos to emulate GitHub login in some headless CMS - in those setups nothing was managed by Netlify, all the rights being handled by GitHub repository access rights, IE if you had push access to the repo you could commit the changes. 

Not a big fan of frontend stuff but could maybe pick this up if you are not rushing too much!

**Comment by nulfrost** [comment]
I can take a look at this, there might need to be some things set up on the netlify end but if that's good to go I'll start working on this as soon as I can

**Comment by nulfrost** [comment]
.take

**Comment by bdougie** [comment]
I setup Netlify Ideentify:

endpoint: https://hot.opensauced.pizza/.netlify/identity

**Comment by bdougie** [comment]
Some more context. Login and Logout should be linked in the nav. This can 

fa-sign-in 
![login](https://user-images.githubusercontent.com/5713670/147365163-ff854ffe-6e37-4598-b7e8-54532b0fc20b.png)

fa-sign-out 
![logout](https://user-images.githubusercontent.com/5713670/147365151-3e39fb2c-86f1-41e0-b84c-dd50d4316df3.png)

**Comment by nulfrost** [comment]
![Screen Shot 2021-12-25 at 1 52 01 PM](https://user-images.githubusercontent.com/16832258/147391668-8fb707ba-5478-4564-b871-5b467ea77759.png)

Making some progress 👍🏿 is there a way to only show the github login button on the widget? doing some google searching but haven't found anything yet

**Comment by bdougie** [comment]
I don't think that is possible without creating a new (fork) widget. Using GoTru directly is what [supabase](https://github.com/netlify/gotrue-js) is doing as well.  

But for the sake of keeping the scope small, leaving the email in, for now, is fine. I may [explore supabase](https://supabase.com/docs/guides/auth) for this in the future to simplify things.

**Comment by nulfrost** [comment]
Hey! So picking up on this again and getting this error when trying to log in with github
![Screen Shot 2022-01-04 at 8 41 50 PM](https://user-images.githubusercontent.com/16832258/148147267-dda3f869-8aec-44b3-b234-31703763bfa6.png)

the link it is referring to leads here https://docs.onegraph.com/docs/authentication.html#asking-your-user-to-sign-in

and another relevant error
![Screen Shot 2022-01-04 at 8 51 59 PM](https://user-images.githubusercontent.com/16832258/148148169-330ae883-9ec5-4fe2-9ec0-6e39eed4705e.png)

As I was typing this I kind of figured maybe the redirect url / callback url are probably configured for the live site and not localhost? if so then I have most of the code already done and would just need to test. Let me know what you think

**Comment by bdougie** [comment]
Sorry about that I gave you the oauth for opensauced.pizza which redirects to the onegraph AuthGuardian service. I just created a new GitHub service specifically for this project. If you try again it should redirect to NEtlify and authorize from there. 

https://docs.netlify.com/visitor-access/oauth-provider-tokens/

![Screen Shot 2022-01-05 at 2 57 54 PM](https://user-images.githubusercontent.com/5713670/148301709-36f1b30a-4861-45ad-8070-33cd8fd9aa22.png)

**Comment by nulfrost** [comment]
Hmm, still seem to be facing errors related to authentication. I've read and double-checked the documentation for the widget but I don't think I'm missing anything 🤔

Here are some of the errors I've encountered
![Screen Shot 2022-01-07 at 7 17 19 PM](https://user-images.githubusercontent.com/16832258/148623217-b73c6173-147b-4871-ba23-c15109e62dc6.png)

and this one is the same from the last post
![Screen Shot 2022-01-07 at 7 17 45 PM](https://user-images.githubusercontent.com/16832258/148623227-87fff25a-e4b9-42b9-bcca-ea613e4702dd.png)

also I'm not sure if this is relevant or not but here's the code I implemented for the github auth stuff
```javascript
import { useState, useEffect } from "react";
import netlifyIdentity from "netlify-identity-widget";

export const useNetlifyAuth = () => {
  const [user, setUser] = useState(null);

  useEffect(() => {
    netlifyIdentity.init();
    setUser(netlifyIdentity.currentUser());
  }, [user]);

  return {
    user,
    login: netlifyIdentity.open,
  };
};

```

I don't think I'm missing anything but here's the branch I'm currently working on too if you wanted to take a look, https://github.com/nulfrost/hot/tree/add-github-login

**Comment by bdougie** [comment]
note: Tried running your branch locally and getting a "esbuild-darwin-arm64" error. I need to remove it to get it to work.

I don't we will be able to use the netlfy identity widget. The requirement to have email is also weird for this use case of github login. I did some quick searching and ran into this thread https://github.com/netlify/gotrue/issues/114#issuecomment-362004553. Basically, we should be leaning on the [go-true](https://github.com/netlify/gotrue) service instead. Luckily supabase already integrates that into its service. The changes implemented  #67 include supabase-js and Netlify. 

Thank you for running through this, but [supabase + github](https://supabase.com/docs/guides/auth/auth-github) is the way to go on this. It will provide us role level security with the DB and has a lot of community support. 

Since the keys are already embedded in the database.js, you can use that to add login

```js
//login session can be added to your custom hook 
async function signInWithGithub() {
  const { user, session, error } = await supabase.auth.signIn({
    provider: 'github',
  })
}

//logout
async function signout() {
  const { error } = await supabase.auth.signOut()
}
```

Ironically, I stumbled on to this article while reseaching all this, which wasn't helpful since it is 5+ years old. https://www.netlify.com/blog/2017/10/30/how-to-add-netlify-identity-service-to-react-projects/ t

**Comment by nulfrost** [comment]
> unrelate note: Tried running your branch locally and getting a "esbuild-darwin-arm64" error. I needed to remove it to get it to work. Curious what that was for. Are you working in a Linux environment that needed that?

Working on an M1 macbook, for some reason without it I wasn't able to get astro to run. Still not sure why lol

> ## netlify-identity will not work
> 
> I don't we will be able to use the netlfy identity widget. The requirement to have email is also weird for this use case of github login. I did some quick searching and ran into this thread [netlify/gotrue#114 (comment)](https://github.com/netlify/gotrue/issues/114#issuecomment-362004553). Basically, we should be leaning on the [go-true](https://github.com/netlify/gotrue) service instead. Luckily supabase already integrates that into its service. The changes implemented #67 include supabase-js and Netlify.
> ## supabase-auth is the solution
> 
> Thank you for running through this, but [supabase + github](https://supabase.com/docs/guides/auth/auth-github) is the way to go on this. It will provide us role-level security with the DB and has a lot of community support.
> 
> Since the keys are already embedded in the [database.js](https://github.com/open-sauced/hot/blob/main/src/lib/database.js#L4-L5), you can use that to add login. You may want to refactor that or leverage your auth files instead.
> 
> ```js
> //login session can be added to your custom hook 
> async function signInWithGithub() {
>   const { user, session, error } = await supabase.auth.signIn({
>     provider: 'github',
>   })
> }
> 
> //logout
> async function signout() {
>   const { error } = await supabase.auth.signOut()
> }
> ```
> 
> Ironically, I stumbled on this article while researching all this, which wasn't helpful since it is 5+ years old. https://www.netlify.com/blog/2017/10/30/how-to-add-netlify-identity-service-to-react-projects/ t

Sweet! I'll take a look at this tonight, I've worked with supabase before so it's all very familiar

**Comment by 0-vortex** [comment]
Make sure you have `node>=14` and `npm@8` and then do this:

```
git restore package.json npm-shrinkwrap.json -s upstream/main
npm ci
npm install --save netlify-identity-widget
```

If you did not clone with `gh` cli, replace `upstream/main` with `origin/main` to get the changes from your own fork.

That should fix all the issues with dependencies! 🍕
